diff --git a/dist/node/serve-DrVGLhCg.js b/dist/node/serve-DrVGLhCg.js
index 48e051436530d9361db13fe37898a98d0e18fe0d..0b7168c2c06eee85e86cb40fa601df545e7dbad5 100644
--- a/dist/node/serve-DrVGLhCg.js
+++ b/dist/node/serve-DrVGLhCg.js
@@ -50279,6 +50279,9 @@ async function serve(options = {}) {
       if (notAnAsset(pathname)) {
         res.setHeader("cache-control", "no-cache");
       }
+      // allow SharedArrayBuffer for `vitepress preview`
+      res.setHeader('Cross-Origin-Opener-Policy', 'same-origin')
+      res.setHeader('Cross-Origin-Embedder-Policy', 'require-corp')
     }
   });
   if (base) {
