diff --git a/dist/node/serve-CXVdC751.js b/dist/node/serve-CXVdC751.js
index bad35030b46e57d42803d5ca7f2402419f03acb9..bc0cbb8db9f1ffcaf6972826c0b52e13999ee63e 100644
--- a/dist/node/serve-CXVdC751.js
+++ b/dist/node/serve-CXVdC751.js
@@ -50279,6 +50279,9 @@ async function serve(options = {}) {
       if (notAnAsset(pathname)) {
         res.setHeader("cache-control", "no-cache");
       }
+      // allow SharedArrayBuffer for `vitepress preview`
+      res.setHeader('Cross-Origin-Opener-Policy', 'same-origin')
+      res.setHeader('Cross-Origin-Embedder-Policy', 'require-corp')
     }
   });
   if (base) {
