diff --git a/dist/node/serve-Csn-CKFx.js b/dist/node/serve-Csn-CKFx.js
index 5815d5154f0fa65ab51b30e7f73fd20ae6a9d9fe..bbd2729011699ab51de7e0d73f6c0ab0b8c2e279 100644
--- a/dist/node/serve-Csn-CKFx.js
+++ b/dist/node/serve-Csn-CKFx.js
@@ -50200,6 +50200,9 @@ async function serve(options = {}) {
       if (notAnAsset(pathname)) {
         res.setHeader("cache-control", "no-cache");
       }
+      // allow SharedArrayBuffer for `vitepress preview`
+      res.setHeader('Cross-Origin-Opener-Policy', 'same-origin')
+      res.setHeader('Cross-Origin-Embedder-Policy', 'require-corp')
     }
   });
   if (base) {
